ick/venv.py

```diff
@@ -33,29 +34,53 @@ def _deps_path(self) -> Path:
         return self.env_path / "deps.txt"
 
     def health_check(self) -> bool:
-        py = self.bin("python")
-        if not py.exists():
-            return False
-        try:
-            _, returncode = run_cmd_status([py, "--version"], check=False)
-        except PermissionError:
-            return False
-        if returncode != 0:
-            return False
-
-        # Eek, this could happen outside the lock, so be defensive against
-        # concurrent modification more than usual
-        try:
-            deps = self._deps_path().read_text()
-        except OSError:
-            return False
-        return deps == json.dumps(self.deps)
-
-    def prepare(self):  # type: ignore[no-untyped-def] # FIX ME
+        # Both None (we don't know) and False (we know it's not working) should
+        # cause us to check again...
+        if not self._cached_health:
```

Comment:
This part needs a little improvement for thundering herd.  The actual creation is protected, but not the health-checking.